### PR TITLE
Fixes #81 : Add landmarks section in side menu of the reader

### LINKED ISSUE #81: EPUB landmarks UI (for list of illustrations, figures, etc.)
This is implemented in `r2-testapp-js`, but is currently missing in `readium-desktop`.
Note that there may be multiple types of landmarks, so this is more like a set of categorized lists.

### PR DESCRIPTION
**Pull request by Scarsniik**

### CODE REVIEW

File: src/renderer/components/reader/ReaderMenu.tsx [hunk 1]
```diff
@@ -74,13 +74,13 @@ export class ReaderMenu extends React.Component<Props, State> {
                 disabled: !publication.TOC || publication.TOC.length === 0,
             },
             {
-                title: __("reader.marks.illustrations"),
-                content: <></>,
-                disabled: !publication.LOI || publication.LOI.length === 0,
+                title: __("reader.marks.landmark"),
```
Comment (llemeurfr) [inline src/renderer/components/reader/ReaderMenu.tsx]: it's landmarks plural. each entry in the set of landmarks is one landmark.

File: src/typings/en.translation.d.ts [hunk 2]
```diff
@@ -273,6 +273,7 @@ declare namespace typed_i18n {
     readonly "annotations": string,
     readonly "bookmarks": string,
     readonly "illustrations": string,
+    readonly "landmark": string,
```
Comment (llemeurfr) [inline src/typings/en.translation.d.ts]: here also, landmarks.

File: src/typings/en.translation.d.ts [hunk 3]
```diff
@@ -328,11 +329,13 @@ declare namespace typed_i18n {
   readonly "annotations": string,
   readonly "bookmarks": string,
   readonly "illustrations": string,
+  readonly "landmark": string,
```
Comment (llemeurfr) [inline src/typings/en.translation.d.ts]: same

File: src/typings/en.translation.d.ts [hunk 4]
```diff
@@ -328,11 +329,13 @@ declare namespace typed_i18n {
...
   readonly "toc": string
 };
   (_: "reader.marks.annotations", __?: {}): string;
   (_: "reader.marks.bookmarks", __?: {}): string;
   (_: "reader.marks.illustrations", __?: {}): string;
+  (_: "reader.marks.landmark", __?: {}): string;
```
Comment (llemeurfr) [inline src/typings/en.translation.d.ts]: same

**Comment by Scarsniik** [comment]
I changed the local name to "landmarks"

**Comment by danielweck** [comment]
To be complete, `any` in the function signature `private createTOCRenderList(TOC: any[]): JSX.Element` should be `Link` (`import { Link } from "r2-shared-js/dist/es6-es2015/src/models/publication-link"`), but I will correct this separately. Let's merge this PR without further delay :)

**Comment by danielweck** [comment]
Hotfix for the missing TypeScript type (replaces "any"):
https://github.com/readium/readium-desktop/commit/6a4306bb9481c1e44614c0b86a0264aec1f06dad

**Pull request merged**